packages/navigator/src/browser/navigator-decorator-service.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2018 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, named } from 'inversify';
+import { Event, Emitter } from '@theia/core/lib/common/event';
+import { ContributionProvider } from '@theia/core/lib/common/contribution-provider';
+import { TreeDecorator, DecorationData, TreeDecoratorService } from '@theia/core/lib/browser/tree/tree-decorator';
+
+/**
+ * Decorator service which emits events from all known tree decorators and caches the current state.
+ */
+@injectable()
+export class NavigatorDecoratorService implements TreeDecoratorService {
```

Comment:
Can we use it as the default implementation or at least put in the tree module? it does not look like it has something specific to the navigator.